GameOfRunes/GameOfRunes/Levels and Waves/LevelCreator.swift

```diff
@@ -0,0 +1,259 @@
+//
+//  LevelCreator.swift
+//  GameOfRunes
+//
+//  Created by Brian Yen on 28/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import Foundation
+
+/**
+ Creates and stored Level-related data.
+ */
+struct LevelCreator {
+    // TODO: Get rid of this once Level Selector is up
+    static func getRandomLevelNumber() -> Int {
+        Int.random(in: 1...3)
+    }
+
+    static func getLevelData(levelNumber: Int) throws -> EnemySpawnUnit {
+        let (levelData, _) = try getLevelDataAndSpawnInterval(levelNumber: levelNumber)
+        return levelData
+    }
+
+    static func getLevelSpawnInterval(levelNumber: Int) throws -> Double {
+        let (_, levelSpawnInterval) = try getLevelDataAndSpawnInterval(levelNumber: levelNumber)
+        return levelSpawnInterval
+    }
+
+    private static func getLevelDataAndSpawnInterval(levelNumber: Int) throws -> (EnemySpawnUnit, Double) {
+        switch levelNumber {
+        case -1:
+            return testLevel
+        case 1:
+            return level1
+        case 2:
+            return level2
+        case 3:
+            return level3
+        default:
+            throw LevelWaveError.invalidLevelNumber
+        }
+    }
+
+    static var level1: (EnemySpawnUnit, Double) {
+        do {
+            let basicOrcUnit = try EnemySpawnUnit(.orc1)
+            let basicTrollUnit = try EnemySpawnUnit(.troll1)
+            let alternatingUnit = basicOrcUnit + basicTrollUnit
+            let fullLevel = alternatingUnit + basicOrcUnit
+            let levelSpawnInterval = 3.0
+            return (fullLevel, levelSpawnInterval)
+        } catch {
+            fatalError("Unexpected error encountered: \(error)")
```

Comment:
Creating errors and catching them with `fatalErrors` seem to not provide any additional functionality except for debugging purposes. Not sure what the others feel about this?

Reply:
As in the reason i threw the errors is because i needed a way to determine that the level isn't valid/cannot be generated. But once this happens, then the game cannot proceed, hence the fatal error

@dongsiji @JermyTan what do yall think.

Reply:
I think it is fine to throw fatal if it is a state the game shouldnt be allowed to enter.

Reply:
I think is fine, alternatively is to just throw a fatalerror when the error occurs instead of propagating up

Reply:
Alrighty :)